proposed/bandwidth-considerations.md

```diff
@@ -0,0 +1,80 @@
+
+
+The MODX Manager is not a website! The assets it loads vary from page to page, user to user, and change as Extras are enabled, installed, and disabled. From a bandwidth perspective, the Manager is the opposite of a web site. As proven in the Proof of Concept section, optimization that commonly apply to websites are anti-optimal when applied to the manager experience.
+
+
+
+
+## Proof of Concept
+Seen as how the pre-2.5 `compress_js` codebase was made under the assumption that developers should reduce the number of HTTP requests at all costs, and nothing else, it was frustrating to be asked to show numbers to prove that this was anti-optimal. That said, some interesting discoveries came from this process. For example, imagine this scenario:  
+
+ - Let's say we have 350kb of scripts that when server side compress_js is used is all in a URL that contains a comma separated list of dependencies like this:
+`/manager/min/index.php?f=/manager/assets/modext/core/modx.localization.js,/manager/assets/modext/util/utilities.js`
+ - Everything is instead in it's own file and possibly not uglified at all (`compress_js` disabled)
+
+
+With these user stories:
+ - `compress_js` enabled
+ - User visits a Resource edit page with `compress_js` enabled
+ - We send them 350kb on script in a single file. `redactor.js` is one of the included files
+ - User disabled Rich Text on the resource and refreshes the page
+ - New URL, so we send them all the scripts again and leverage the cache for none of them even though all we are doing is sending them same scripts minus redactor
+
+
+• User visits a Resource edit page with compress_js disabled
+	• We send them 350kb or more of script in individual files. redactor.js is one of the included files
+	• User disabled Rich Text on the resource and refreshes the page
+
+
+Scenario A has a cache usage of 0%. Even though we are serving less data by disabling an Extra we resend all the bytes. The URL changes, the cache is purged. We couldn’t leverage the browser cache less if we tried. We couldn’t make the experience more expensive in terms of bandwidth if we tried.
+
+
+Scenario B has a cache usage of 100%. Each dependency is its own request, and therefore own URL. The cache is leveraged as much as possible. Yes this means more HTTP requests and a possibly slower load time (probably not) over HTTP/1 but that only applies to the first page load. Once assets are cached, they are cached!
+
+
+
+
+
+## Considerations
+The cost of bandwidth varies greatly in different countries. For example, the average Brazilian works over 34 hours for every 500MB of bandwidth. By being considerate of bandwidth, we make the MODX site building more accessible to our global community.
+
+![](http://j4p.us/0S0e2g1F2r0v/Image%202016-11-03%20at%2010.46.24%20PM.jpg)
```

Comment:
I don't think recommendations should contain an image hosted elsewhere, plus the data in that image is not accessible.